cognite_toolkit/cognite_modules/examples/cdf_data_pipeline_files_valhall/functions/fu_context_files_oid_fileshare_annotation/pipeline.py

```diff
@@ -0,0 +1,554 @@
+        # Create annotations once we hit 1k (to spread insertion over time):
+        if len(create_annotation_list) == 1000:
+            client.annotations.create(create_annotation_list)
+            create_annotation_list.clear()
+
+    client.annotations.create(create_annotation_list)
+    safe_delete_annotations(to_delete_annotation_list, client)
+    # De-duplicate list of names and id before returning:
+    return list(set(entities_name_found)), list(set(entities_id_found))
+
+
+def retrieve_diagram_with_retry(
+    client: CogniteClient, entities: list[Entity], file_id: str, retries: int = 3
+) -> DiagramDetectResults:
+    for retry_num in range(1, retries + 1):
+        try:
+            return client.diagrams.detect(
+                file_external_ids=[file_id],
+                search_field="name",
+                entities=[e.dump() for e in entities],
+                partial_match=True,
+                min_tokens=2,
+            )
+        except Exception as e:
+            # retry func if CDF api returns an error
+            if retry_num < 3:
+                print(f"[WARNING] Failed to detect entities, retry #{retry_num}, error: {type(e)}({e})")
+                time.sleep(retry_num * 5)
+            else:
+                msg = f"Failed to detect entities, error: {type(e)}({e})"
+                print(f"[ERROR] {msg}")
+                raise RuntimeError(msg)
+
+
+def get_sys_nums(annotations: Any, detected_count: int) -> tuple[dict[str, int], int]:
+    """Get dict of used system number in P&ID. The dict is used to annotate if system
+    number is missing - but then only annotation of found text is part of most
+    frequent used system number
+
+    Args:
+        annotations: list of annotations found by context api
+        detected_count: total number of detected system numbers
+
+    Returns:
+        tuple[dict[str, int], int]: dict of system numbers and number of times used
+    """
+    detected_sytem_num = defaultdict(int)
+    for item in annotations:
+        tokens = item["text"].split("-")
+        if len(tokens) == 3:
+            detected_count += 1
+            detected_sytem_num[tokens[0]] += 1
+
+    return dict(detected_sytem_num), detected_count
+
+
+def get_coordinates(vertices: list[dict]) -> dict[str, int]:
+    """Get coordinates for text box based on input from contextualization
+    and convert it to coordinates used in annotations.
+
+    Args:
+        vertices (list[dict]): coordinates from contextualization
+
+    Returns:
+        dict[str, int]: coordinates used by annotations.
+    """
+    x_min, *_, x_max = sorted(min(1, vert["x"]) for vert in vertices)
+    y_min, *_, y_max = sorted(min(1, vert["y"]) for vert in vertices)
+
+    # Adjust if min and max are equal
+    if x_min == x_max:
+        x_min, x_max = x_min - 0.001, x_max if x_min > 0.001 else x_min, x_max + 0.001
+    if y_min == y_max:
+        y_min, y_max = y_min - 0.001, y_max if y_min > 0.001 else y_min, y_max + 0.001
```

Comment:
```suggestion
    if x_min == x_max:
        x_min, x_max = (x_min - 0.001, x_max) if x_min > 0.001 else (x_min, x_max + 0.001)
    if y_min == y_max:
        y_min, y_max = (y_min - 0.001, y_max) if y_min > 0.001 else (y_min, y_max + 0.001)
```
If not you are unpacking 3 values, i.e., `x_min-0.001, (x_max if x_min > 0.001 else x_min), x_max +0.001`

Reply:
🤦‍♂️ , thanks for catching! Seems like my fav activity lately is deleting necessary brackets 😂